packages/zoe/test/unitTests/contracts/test-priceAggregator.js

```diff
@@ -208,57 +255,118 @@ test('median aggregator', /** @param {ExecutionContext} t */ async t => {
     return { timestamp, amountOut: valueOut };
   };
 
-  const pricePushAdmin = await E(aggregator.creatorFacet).initOracle(
-    pricePush.instance,
-  );
+  await publications.nextMatches({ amountOut: 1010n, timestamp: 0n });
 
   const quote0 = await tickAndQuote();
   t.deepEqual(quote0, { amountOut: 1020n, timestamp: 1n });
+  await publications.nextMatches(quote0);
 
   const quote1 = await tickAndQuote();
   t.deepEqual(quote1, { amountOut: 1030n, timestamp: 2n });
+  await publications.nextMatches(quote1);
 
   const price1300Admin = await E(aggregator.creatorFacet).initOracle(
     price1300.instance,
     {
       increment: 8n,
     },
   );
+  await publications.nextMatches({
+    // quoteNotifier is for 1n but some results come back for a greater amountIn
+    amountIn: 2n,
+    amountOut: 2338n,
+    timestamp: 2n,
+  });
 
   const quote2 = await tickAndQuote();
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 2348n,
+    timestamp: 3n,
+  });
   t.deepEqual(quote2, { amountOut: 1178n, timestamp: 3n });
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 2356n,
+    timestamp: 3n,
+  });
 
   const quote3 = await tickAndQuote();
   t.deepEqual(quote3, { amountOut: 1187n, timestamp: 4n });
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 2366n,
+    timestamp: 4n,
+  });
 
   await E(aggregator.creatorFacet).initOracle(price800.instance, {
     increment: 17n,
   });
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 2374n,
+    timestamp: 4n,
+  });
+  await publications.nextMatches({
+    amountIn: 1n,
+    amountOut: 1050n,
+    timestamp: 4n,
+  });
 
   const quote4 = await tickAndQuote();
   t.deepEqual(quote4, { amountOut: 1060n, timestamp: 5n });
+  await publications.nextMatches(quote4);
 
   const quote5 = await tickAndQuote();
   t.deepEqual(quote5, { amountOut: 1070n, timestamp: 6n });
+  await publications.nextMatches(quote5);
 
   // Push a price into the fray.
+  const pricePushAdmin = await E(aggregator.creatorFacet).initOracle(
+    pricePush.instance,
+  );
   await E(pricePushAdmin).pushResult('1069');
 
   const quote6 = await tickAndQuote();
   t.deepEqual(quote6, { amountOut: 1074n, timestamp: 7n });
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 2149n,
+    timestamp: 7n,
+  });
 
   await E(pricePushAdmin).delete();
+  await publications.nextMatches({
+    amountIn: 1n,
+    amountOut: 1080n,
+    timestamp: 7n,
+  });
 
   const quote7 = await tickAndQuote();
   t.deepEqual(quote7, { amountOut: 1090n, timestamp: 8n });
+  await publications.nextMatches({
+    amountIn: 1n,
+    amountOut: 1090n,
+    timestamp: 8n,
+  });
 
   await E(price1300Admin).delete();
+  await publications.nextMatches({
+    amountIn: 2n,
+    amountOut: 1975n,
+    timestamp: 8n,
+  });
 
   const quote8 = await tickAndQuote();
   t.deepEqual(quote8, { amountOut: 1001n, timestamp: 9n });
+  await publications.nextMatches({
```

Comment:
1001n (just above)  seems to be wrong.  It gets that result by combining the 9th tick from `price1000` with the 10th tick from `price800`. I don't yet see why that happens.

The result your iterator gets for tick 9 (1985/2)  is correct from my calculations. The 9th tick for `price1000` is 1100, and the 9th tick for `price800` is 885.

Reply:
Okay, it's not a bug you created or made worse.  Would you mind adding a comment on line 361 referencing #5851?  And you should expect the publication's result to continue to be ` 1985n / 2n `.